controllers/machine_controller_phases.go

```diff
@@ -197,6 +198,14 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, m *clus
 	if infraConfig == nil && err == nil {
 		return nil
 	} else if err != nil {
+		if m.Status.InfrastructureReady && strings.Contains(err.Error(), "could not find") {
+			// Infra object went missing after the machine was up and running
+			r.Log.Error(err, "Machine infrastructure reference has been deleted after being ready, setting failure state")
+			machineStatusError := capierrors.InvalidConfigurationMachineError
+			m.Status.ErrorReason = &machineStatusError
```

Comment:
```suggestion
			m.Status.ErrorReason = capierrors.ClusterStatusErrorPtr(capierrors.InvalidConfigurationMachineError)
```

Reply:
> 

Didn't know this exist. Shouldn't it be MachineStatusErrorPtr instead of ClusterStatusErrorPtr